beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java

```diff
@@ -100,12 +102,13 @@
     }
   }
 
+  @SuppressWarnings("unused")
   @Value
-  private final class BatchSnapshotDescriptor implements PCollectionDescriptor {
-    private final Pipeline pipeline;
-    private final DataAccessor dataAccessor;
-    private final long fromStamp;
-    private final long untilStamp;
+  private static class BatchSnapshotDescriptor implements PCollectionDescriptor {
```

Comment:
Do we want to remove the `final` modifier?

Reply:
It's already `final` by @Value annotation from Lombok - https://projectlombok.org/features/Value

Reply:
ah, yep.